slothy/targets/aarch64/aarch64_neon.py

```diff
@@ -49,6 +51,111 @@ class which generates instruction parsers and writers from instruction templates
 arch_name = "Arm_AArch64"
 llvm_mca_arch = "aarch64"
 
+class AArch64CheckerFailure(Exception):
+    """Exception thrown when something goes wrong during equivalence testing"""
+
+class AArch64CheckerNotAvailable(Exception):
+    """Exception thrown when the checker cannot run (because of an unsupported
+    platform or configuration, for example)"""
+
+class Checker:
+    """Empirical equivalence checker for AArch64 kernels
+
+    This is a Python wrapper around the equivalence checker in
+    tests/aarch64/checker. It takes as argument an instance of
+    the `Result` class and tests equivalence of input and output
+    by compiling and running it, checking that registers marked
+    as output are indeed always the same.
+
+    The checker also tries to deal with pointers and memory contents,
+    but needs to be instructed ot know how much memory is assumed to be
+    present, and which GPRs reference it.
+    """
+
+    @staticmethod
+    def check(result, address_gprs, log):
+        """Run an equivalence test on an instance of the Result class
+
+        Arguments:
+         - result: Instance of Result class representing the SLOTHY run
+                   to be checked. This includes both the optimized and the
+                   original source code.
+         - address_gprs: GPR-indexed dictionary indicating which registers
+                   contain pointers to memory. The value indicates the size
+                   of the memory. For example, { "x0" : 1024 } would lead to
+                   X0 being set to the base of a 1024 byte buffer."""
+
+        # For now, we only support native AArch64 systems
+        if platform.machine() != "arm64":
```

Comment:
```suggestion
        if platform.machine() not in ["arm64", "aarch64"]:
```

On my Raspberry Pi 4, `platform.machine()` returns `aarch64`.

Reply:
No longer relevant after the switch to unicorn.